pkg/provisioner/ironic/ironic.go

```diff
@@ -1937,8 +1937,9 @@ func (p *ironicProvisioner) PowerOff(rebootMode metal3v1alpha1.RebootMode, force
 
 	if ironicNode.PowerState != powerOff {
 		targetState := ironicNode.TargetPowerState
+		useSoftReboot := rebootMode == metal3v1alpha1.RebootModeSoft
 		// If the target state is either powerOff or softPowerOff, then we should wait
-		if targetState == powerOff || targetState == softPowerOff {
+		if targetState == powerOff || (useSoftReboot && targetState == softPowerOff) {
```

Comment:
Not that it's wrong, but I don't quite understand the point of the change. If `useSoftReboot` is false, `targetState` will never be `softPowerOff`

Reply:
`targetState` is being read from the node in ironic, so it could have been set differently by a previous call to `PowerOff()`. e.g. if a soft power off was in progress ("please give consideration to possibly rebooting this host") but a hard power off request comes in ("fence this node, now") then previously we would have kept waiting to see if the first one worked.

The thing to be sure of is that ironic can handle this sequence of events.

Reply:
I think Ironic will hold a lock that will prevent you from issuing another request immediately.

Reply:
:disappointed: 
Is there anything we can do from the Metal³ side that would be correct?